src/promptflow-core/promptflow/core/_prompty_utils.py

```diff
@@ -1162,6 +1163,114 @@ def wrapper(*args, **kwargs):
     return decorator
 
 
+def handle_openai_error_async(tries: int = 10, unprocessable_entity_error_tries: int = 3):
```

Comment:
I tried to merge this code with the existing handle_openai_error decorator, but found it difficult. So, I created a separate decorator for the async version to keep the code clean